src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -1957,51 +1982,34 @@ JNI_METHOD(void, write)
         VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
         if (tlvBytesObj != nullptr)
         {
-            jbyte * tlvBytesObjBytes = env->GetByteArrayElements(tlvBytesObj, nullptr);
-            VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
-            length = static_cast<size_t>(env->GetArrayLength(tlvBytesObj));
-            VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
-            tlvBytes    = reinterpret_cast<uint8_t *>(tlvBytesObjBytes);
-            hasValidTlv = true;
+            JniByteArray tlvByteArray(env, tlvBytesObj);
+            app::ConcreteDataAttributePath path(static_cast<EndpointId>(endpointId), static_cast<ClusterId>(clusterId),
```

Comment:
path seems identical between both if branches. Would it make sense to extract it outside the If?